.github/workflows/ci.yml

```diff
@@ -0,0 +1,41 @@
+name: Regular CI Pipeline
+
+on:
+  pull_request:
+    branches:
+      # Regular release channels
+      - master
+      - next
+      - beta
+      - alpha
+      # Support, hotfix branches like: 1.0.x or 1.x
+      - "+([0-9])?(.{+([0-9]),x}).x"
+
+  # Allows triggering the workflow manually
+  workflow_dispatch:
+
+jobs:
+  regular-checks:
+    name: Build and unit-test on supported platforms and NodeJS versions
+    strategy:
+      matrix:
+        node-version: [16.x, 18.x, 20.x]
+        os: [ubuntu-latest, windows-latest, macos-latest]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup NodeJS ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Perform regular checks
+        run: |
+          npm install
+          npm run format:check
+          npm run lint
+          npm run test:unit
+          npm run build
```

Comment:
Maybe we should do a build before units first? if TSC doesn't pass, there's no need to run tests..

Reply:
As per Discord -> test:unit will force tsc due to `ts-jest`, so transpilation errors will fail the build early.

Reply:
You're right